File: src/main/scala/examples/mixins/Example3.scala
package examples.mixins

trait Example3 extends Example3Mixin1 with Example3Mixin2

private[mixins] trait Example3Mixin1 {
  def plus(a: Int, b: Int): Int = {
    a + b
  }
}

private[mixins] trait Example3Mixin2 {
  def times(a: Int, b: Int): Int = {
    a * b
  }
}

private[mixins] class Example3Impl extends Example3

File: src/main/java/controls/mixins/Example3.java
package controls.mixins;



    interface Example3 extends Example3Mixin1, Example3Mixin2 {
    }

    interface Example3Mixin1 {
        default int plus(int a, int b) {
            return a + b;
        }
    }

    interface Example3Mixin2 {
        default int times(int a, int b) {
            return a * b;
        }
    }



class Example3Impl implements Example3 {
}